spark/common/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala

```diff
@@ -1435,6 +1436,58 @@ case class ST_ForceRHR(inputExpressions: Seq[Expression])
   }
 }
 
+case class ST_GeneratePoints(inputExpressions: Seq[Expression], randomSeed: Long)
+    extends Expression
+    with CodegenFallback
+    with ExpectsInputTypes
+    with Nondeterministic {
+
+  def this(inputExpressions: Seq[Expression]) = this(inputExpressions, Utils.random.nextLong())
+
+  @transient private[this] var random: java.util.Random = _
+
+  private val nArgs = children.length
+
+  override protected def initializeInternal(partitionIndex: Int): Unit = random =
+    new java.util.Random(randomSeed + partitionIndex)
+
+  override protected def evalInternal(input: InternalRow): Any = {
+    val geom = children.head.toGeometry(input)
+    val numPoints = children(1).eval(input).asInstanceOf[Int]
+    val generatedPoints = if (nArgs == 3) {
+      val seed = children(2).eval(input).asInstanceOf[Int]
+      if (seed > 0) {
+        Functions.generatePoints(geom, numPoints, seed)
+      } else {
+        Functions.generatePoints(geom, numPoints, random)
```

Comment:
Why is this needed? This if condition seems to be repeating the same condition in `Functions`

Reply:
This is for generating consistent random streams even if seed is 0.